### PR TITLE
Item 6: Validate denom on Manager contract

### PR DESCRIPTION
**Pull request by mikedotexe**
We use the query for the bonded denom to validate whether the provided denom is valid.

There were a few cases across other contracts that used hardcoded values that then needed to be updated, as well.

UPDATE: removed the `denom` from the Manager's instantiate message as discussed in the comments here. Also modified the CosmJS deploy script accordingly.

### CODE REVIEW

File: contracts/croncat-manager/src/tests/manager_tests.rs [hunk 1]
```diff
@@ -128,14 +125,14 @@ mod instantiate_tests {
...
-            native_denom: "cron".to_owned(),
+            native_denom: DENOM.to_string(),
             limit: 100,
             treasury_addr: Some(Addr::unchecked(AGENT2)),
         };
         assert_eq!(config, expected_config);
 
         let manager_balances = query_manager_balances(&app, &manager_addr);
-        assert_eq!(manager_balances, Uint128::new(600));
+        assert_eq!(manager_balances, Uint128::new(2400));
```
Comment (mikedotexe) [inline contracts/croncat-manager/src/tests/manager_tests.rs:135]: Note that I took some time to make sure this wasn't just "changing the test to match the output" and indeed verified this tracks.

File: contracts/croncat-agents/src/tests/common.rs [hunk 2]
```diff
@@ -30,7 +30,9 @@ pub const PARTICIPANT5: &str = "cosmos5t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd";
 pub const PARTICIPANT6: &str = "cosmos6t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd";
 pub const PARTICIPANT7: &str = "cosmos7t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd";
 
-pub const NATIVE_DENOM: &str = "uatom";
+/// We set this to "TOKEN" to match the denom here:
+/// https://github.com/CosmWasm/cosmwasm/blob/32f308a1a56ae5b8278947891306f7a374c3df94/packages/vm/src/environment.rs#L383
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/tests/common.rs:34]: ah nice, was wondering about that!

File: contracts/croncat-manager/src/contract.rs [hunk 3]
```diff
@@ -63,6 +63,16 @@ pub fn instantiate(
         cw20_whitelist,
     } = msg;
 
+    // Validate the denom, verify as case-insensitive, use bonded_denom value
+    let bonded_denom = deps.querier.query_bonded_denom()?;
+    let denom = if denom.to_uppercase() != bonded_denom.to_uppercase() {
```
Comment (TrevorJTClarke) [inline contracts/croncat-manager/src/contract.rs:68]: @mikedotexe the goal is to actually remove instantiate denom, so we only rely on the protocol native denom

File: contracts/croncat-manager/src/tests/manager_tests.rs (hunk 1 above)
Comment (TrevorJTClarke) [reply contracts/croncat-manager/src/tests/manager_tests.rs:135]: I see the value above, fine w / me

File: contracts/croncat-manager/src/tests/manager_tests.rs [hunk 4]
```diff
@@ -4698,3 +4695,70 @@ fn immediate_event_task_has_multiple_executions() {
...
+
+/// We test that the denom provided matches the validated, bonded denom value
+#[test]
+fn invalid_denom() {
+    let mut app = default_app();
+    let factory_addr = init_factory(&mut app);
+
+    let mut instantiate_msg: InstantiateMsg = default_instantiate_message();
+    // Test different, invalid denom
+    instantiate_msg.denom = "mochi".to_string();
```
Comment (TrevorJTClarke) [inline contracts/croncat-manager/src/tests/manager_tests.rs:4707]: You did good work on this test - but as per above, i think we shouldn't use instantiate denom

**Review by TrevorJTClarke — CHANGES_REQUESTED** [review]
Thanks for doing this!! See comments:

File: contracts/croncat-manager/src/contract.rs (hunk 3 above)
Comment (mikedotexe) [reply contracts/croncat-manager/src/contract.rs:68]: Ah, gotcha

**Review by TrevorJTClarke — APPROVED** [review]
Brilliant!

**Pull request merged**